BlazarUI/app/scripts/components/module/BuildButton.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {Component, PropTypes} from 'react';
+import {Button} from 'react-bootstrap';
+import Icon from '../shared/Icon.jsx';
+import $ from 'jquery';
+
+class BuildButton extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {loading: false, textStatus: ''};
+  }
+
+  build() {
+    this.setState({
+      loading: true
+    });
+    $.ajax({
+      url: `${config.apiRoot}/build/module/${this.props.moduleId}`,
+      type: 'POST'
+    }).then(() => {
+      this.setState({
+        loading: false,
+        textStatus: ''
+      });
+    },
+    (data, textStatus, jqXHR) => {
+      this.setState({
+        loading: false,
+        textStatus: jqXHR
+      });
+    });
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <Button bsStyle="primary" disabled><Icon type="fa" name="circle-o-notch" classNames="fa-spin" /> Build Now</Button>;
```

Comment:
I may have done some one liners in the past, but moving forward let's try to always return markup wrapped in parents and indented to improve readability. e.g. -
```javascript
return (
   <Button bsStyle="primary" disabled>
     <Icon type="fa" name="circle-o-notch" classNames="fa-spin" /> Build Now
  </Button>
);
```